DNN Platform/DotNetNuke.Abstractions/Portals/IPortalSettings.cs

```diff
@@ -370,5 +370,10 @@ public interface IPortalSettings
 
         /// <summary>Gets a value indicating whether to display the dropdowns to quickly add a moduel to the page in the edit bar.</summary>
         bool ShowQuickModuleAddMenu { get; }
+
+        /// <summary>
+        /// Gets the dnn styles for this portal, these get injected on the page top as css custom properties.
+        /// </summary>
+        IPortalStyles Styles { get; }

```

Comment:
IMO we should try to get it as right as we can and attempt to prevent changes, but when it is inevitable, I tend to like a version. With modern tooling, the name gets autocompleted anyway and I have seen that practice in some places including DirectX `IDirect3DDevice9` for v9; Windows COM 64 bit APIs, Android `Camera2` APIs when they revamped the whole thing.

I think this approach would allow us to keep the old interface (deprecated) for 2 major versions.

Now we should probably not do this very often and still try to think the initial interface through, we can also use extension methods if we are just adding some utility to an existing interface, etc.

Reply:
I think this is the only remaining issue I have with this PR. Do we call this type `IPortalSettings2` or `IPortalSettings10` (and deprecate `IPortalSettings` for DNN 12)?

Thoughts @dnnsoftware/approvers?

Reply:
I kinda like the idea of using the major version, so IPortalSettings10 IMO. I guess it would make it easier to know in which DNN version the new interface started existing.

Now should we duplicate the existing interface or inherit from it ?

Reply:
I want to say inherit, but does that mean that it would inherit the `Obsolete` attribute?

Reply:
Yeah, I am not sure, let me try that out...

Reply:
With this:
![image](https://github.com/dnnsoftware/Dnn.Platform/assets/6371568/10577641-f0ca-4928-9f05-7c361396d7f3)

It appears to be fine:
![image](https://github.com/dnnsoftware/Dnn.Platform/assets/6371568/653d3ffd-ed22-4403-a5a8-bc468ddcb510)

I'll go with that approach...

Reply:
That has a lot of ramifications though :)
![image](https://github.com/dnnsoftware/Dnn.Platform/assets/6371568/f98c4256-56b8-45ca-aaca-64d74e806687)

Which I can handle, but just want to make sure that creating new overloads and deprecating the old methods is the proper way...

Reply:
Another approach could be to not make this part of IPortalSettings at all... Open to suggestions.

Reply:
One idea I can think would be to create an extension method so consumers woudl do:

portalSettings.GetStyles()

does that sound like a good idea to you ?

Reply:
Could also create IPortalSettings10 but not deprecate IPortalSettings, those who want the new features just use the new interface...

Reply:
@bdukes Ok, I removed the change on IPortalSettings and created an extension method. Please see https://github.com/dnnsoftware/Dnn.Platform/pull/5851/commits/51dce1e05684ea210c6eb7b553d319875967d535 and let me know what you thing, we can revert that commit if we come up with a better idea. But this way we don't change the interface at all.

Reply:
I like this approach @valadas (extension method).

Reply:
Looks good to me, thanks!